### DISCUSSION TITLE
--add-modules javafx.base to "Runtime Args" in Publish Tool

### DISCUSSION THREAD

**Discussion by timoreichen** [shannah/jdeploy#85] [Q&A]
I can publish my app but it doesnt start up on any platform.
My guess that --add-modules is not properly added. 

The app can be launched from commandline like this:

java --add-modules javafx.base,javafx.graphics,javafx.fxml,javafx.media,javafx.web --add-exports javafx.base/com.sun.javafx.event=ALL-UNNAMED --add-exports javafx.graphics/com.sun.javafx.scene=ALL-UNNAMED --add-exports javafx.graphics/com.sun.javafx.scene.traversal=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-exports javafx.web/com.sun.javafx.webkit=ALL-UNNAMED -jar target/Rocketbatch-1.0.jar

How adding --add-modules etc within the publish tool as "Runtime Args". 
Thanks for any help!

**Comment by shannah** [comment]
See [this example app's package.json file](https://github.com/amitjoy/osgifx/blob/main/package.json#L11-L23) which adds these runtime args via the package.json file config:

```json
...
"jdeploy": {
        ...
        "args": [
            "--add-exports javafx.base/com.sun.javafx.event=ALL-UNNAMED",
            "--add-exports javafx.base/com.sun.javafx.runtime=ALL-UNNAMED",
            "--add-exports javafx.graphics/com.sun.glass.ui=ALL-UNNAMED",
            "--add-exports javafx.graphics/com.sun.javafx.scene=ALL-UNNAMED",
            "--add-exports javafx.graphics/com.sun.javafx.application=ALL-UNNAMED",
            "--add-exports javafx.graphics/com.sun.glass.ui=ALL-UNNAMED",
            "--add-exports javafx.graphics/com.sun.javafx.css=ALL-UNNAMED",
            "--add-opens javafx.graphics/javafx.scene=ALL-UNNAMED",
            "--add-opens javafx.controls/javafx.scene.control.skin=ALL-UNNAMED"
        ],
...
```

**Reply by timoreichen** [reply]
Thanks for the Example. I double checked the package.json, it was already correct as in the example. So my guess was wrong its the argumentlist. Do you have any other idea maybe what else goes wrong when startup the app? The app itself doesn't start to log itself, so the error must be on launch.

**Reply by shannah** [reply]
Can you share the download link?

**Reply by timoreichen** [reply]
thats: https://www.jdeploy.com/~rocketbatch

**Reply by shannah** [reply]
Here's the error I get

``` bash
./Contents/MacOS/Client4JLauncher 
Unrecognized option: --add-modules javafx.base,javafx.graphics,javafx.fxml,javafx.media,javafx.web
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

Does that help you track it down?

**Reply by timoreichen** [reply]
Yep. Thanks for your help!! And great work, imho the most important project(deployment) for the success of java(fx)!!
